suite-common/wallet-core/src/discovery/discoveryThunks.ts

```diff
@@ -74,44 +73,32 @@ export const filterUnavailableAccountTypes = (
             !device?.unavailableCapabilities?.[networkAccount.accountType!], // exclude by account types (ex: taproot)
     );
 
-/**
- * @deprecated Old implementation, temporarily duplicated here for suite, while the new one is used in suite-native
- * Fully replaced by combined use of filterUnavailableNetworks and filterUnavailableAccountTypes
- */
-export const filterUnavailableNetworksCompatible = (
-    enabledNetworks: NetworkSymbol[],
-    device?: TrezorDevice,
-) =>
-    networksCompatibility.filter(n => {
-        const firmwareVersion = getFirmwareVersion(device);
-        const internalModel = device?.features?.internal_model;
-
-        const isSupportedInSuite =
-            !n.support || // support is not defined => is supported
-            !internalModel || // typescript. device undefined. => supported
-            (n.support[internalModel] && // support is defined for current device
-                versionUtils.isNewerOrEqual(firmwareVersion, n.support[internalModel])); // device version is newer or equal to support field in networks => supported
-
-        return (
-            isTrezorConnectBackendType(n.backendType) && // exclude accounts with unsupported backend type
-            enabledNetworks.includes(n.symbol) &&
-            !n.isHidden &&
-            !device?.unavailableCapabilities?.[n.accountType!] && // exclude by account types (ex: taproot)
-            !device?.unavailableCapabilities?.[n.symbol] && // exclude by network symbol (ex: xrp on T1B1)
-            isSupportedInSuite
-        );
-    });
-
 const calculateProgress =
     (discovery: Discovery) =>
     (_dispatch: any, getState: any): PartialDiscovery => {
-        const { numberOfNonCardano, numberOfCardano } = discovery.networks.reduce(
-            (acc, symbol) => {
-                if (symbol === 'ada') {
-                    acc.numberOfCardano += 1;
-                } else {
-                    acc.numberOfNonCardano += 1;
-                }
+        const device = selectDevice(getState());
+        // reconstruct networks from discovery symbols, because we need to iterate through accounts
+        const networksToCount = filterUnavailableNetworks(discovery.networks, device);
+
+        // number of Cardano type coins which are activated and able to be discovered
+        const numberOfCardanoCoins = discovery.networks.filter(
+            symbol => symbol === 'ada' || symbol === 'tada',
+        ).length;
+
+        // numbers of accounts of givent types
+        const { numberOfNonCardano, numberOfCardano } = networksToCount.reduce(
+            (acc, network) => {
+                const { symbol } = network;
+
+                // increment the appropriate counter based on the network symbol
+                const incrementAccumulator = () => {
+                    if (symbol === 'ada' || symbol === 'tada') acc.numberOfCardano += 1;
```

Comment:
Note: at first I thought just including `'tada'` here would solve the :tada: :bug:, but I also had to do the 1 or 2 thing instead of constant 1.
Otherwise, with `ada` & `tada` at the same time, discovery was wrongly indicated as finished, with debug showing `16 / 6` as loaded / total. I would have to activate another coins to start the discovery of `ada` or `tada`

Reply:
Thank you! (Maybe I am just blind but it would be worth it to add tests for that?)